src/vs/platform/extensions/common/extensions.ts

```diff
@@ -137,6 +137,7 @@ export interface IWalkthroughStep {
 export interface IWalkthrough {
 	readonly id: string;
 	readonly title: string;
+	readonly icon?: string;
```

Comment:
This is a great start. For the future though, we may also need to support light and dark icon variants

Reply:
Given this is an extension point contribution should we do that now to avoid further changes to the entrypoint or do we think it's fine since it's likely to just be an or type such as `icon?: string | {light: string, dark: string}`

Reply:
I don't think I'd be able to get that change soon in since it's a little bit more involved (plus, it brings in some questions about whether extension icons can support multiple themes too). I think you're right that the declaration will likely be something like

```ts
icon?: string | { light: string, dark: string, hc?: string, hcLight?: string };
```

so we can expand in the future.